.eslintrc.js

```diff
@@ -0,0 +1,53 @@
+module.exports = {
+  parser: '@typescript-eslint/parser', // Specifies the ESLint parser
+  // plugins: [
+  //   '@typescript-eslint',
+  // ],
+  parserOptions: {
+    ecmaVersion: 2020, // Allows for the parsing of modern ECMAScript features
+    sourceType: 'module', // Allows for the use of imports
+    ecmaFeatures: {
+      jsx: true // Allows for the parsing of JSX
+    }
+  },
+  settings: {
+    react: {
+      version: 'detect' // Tells eslint-plugin-react to automatically detect the version of React to use
+    }
+  },
+  extends: [
+    'plugin:react/recommended', // Uses the recommended rules from @eslint-plugin-react
+    'plugin:@typescript-eslint/recommended', // Uses the recommended rules from @typescript-eslint/eslint-plugin
+    "prettier/@typescript-eslint", // Uses eslint-config-prettier to disable ESLint rules from @typescript-eslint/eslint-plugin that would conflict with prettier
+    "plugin:prettier/recommended", // Enables eslint-plugin-prettier and eslint-config-prettier. This will display prettier errors as ESLint errors. Make sure this is always the last configuration in the extends array.
+    'react-app', // Uses recommended rules from react-scripts
+  ],
+  rules: {
+    // Place to specify ESLint rules. Can be used to overwrite rules specified from the extended configs
+    // eslint rules: https://eslint.org/docs/rules/
+    // @typescript-eslint rules: https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin
+    '@typescript-eslint/explicit-function-return-type': 'off', // with this we won't need to specify functions return types. Typescript will infer it for us
+    'no-console': 'warn', // Lets not leave console.logs all over
+    curly: 'warn', // forces { and } after ifs, while, for, etc
+    indent: ['warn', 2],
```

Comment:
I prefer tab indentation instead of two spaces, but this is what I saw everywhere I looked. I'm open to opinions.

Reply:
my vote is two spaces

Reply:
I configured it for two spaces